customcollection.go

```diff
@@ -0,0 +1,96 @@
+package goshopify
+
+import (
+	"fmt"
+)
+
+const customCollectionsBasePath = "admin/custom_collections"
+
+// CustomCollectionService is an interface for interacting with the custom
+// collection endpoints of the Shopify API.
+// See https://help.shopify.com/api/reference/customcollection
+type CustomCollectionService interface {
+	List(interface{}) ([]CustomCollection, error)
+	Count(interface{}) (int, error)
+	Get(int, interface{}) (*CustomCollection, error)
+	Create(CustomCollection) (*CustomCollection, error)
+	Update(int, CustomCollection) (*CustomCollection, error)
+	Delete(int) error
+}
+
+// CustomCollectionServiceOp handles communication with the custom collection
+// related methods of the Shopify API.
+type CustomCollectionServiceOp struct {
+	client *Client
+}
+
+// CustomCollection represents a Shopify custom collection.
+type CustomCollection struct {
+	ID             int    `json:"id"`
+	Handle         string `json:"handle"`
+	Title          string `json:"title"`
+	UpdatedAt      string `json:"updated_at"`
+	BodyHTML       string `json:"body_html"`
+	SortOrder      string `json:"sort_order"`
+	TemplateSuffix string `json:"template_suffix"`
+	Image          Image  `json:"image"`
```

Comment:
According to the docs, this is [actually a `string`](https://help.shopify.com/api/reference/customcollection#properties). I haven't worked with this collection in a while, so I can't verify. Do you know what the general return type is? Sometimes the docs are not fully updated.

Reply:
Docs are wrong on that one for some reason. I've thrown an example object below from the API.

```json
        {
            "id": (removed-int-id),
            "handle": "example-manual-collection",
            "title": "Example Manual Collection",
            "updated_at": "2018-02-06T01:48:03-05:00",
            "body_html": "Example body.",
            "published_at": "2018-02-06T01:42:43-05:00",
            "sort_order": "best-selling",
            "template_suffix": null,
            "published_scope": "web",
            "image": {
                "created_at": "2018-02-06T01:43:10-05:00",
                "width": 3000,
                "height": 2000,
                "src": "(removed image URL)"
            }
        },
```

Reply:
Thanks for the example 👍